### ISSUE TITLE
Static OpenCV doesn't read LZW TIFFs

### ISSUE THREAD

**Issue by csparker247** [Itseez/opencv#5662]
I've been attempting to build a static version of OpenCV on OSX 10.10 and have been having trouble reading LZW compressed TIFF images using imread. Specifically, the resulting Mat is empty (null data member). Uncompressed TIFFs read with no problems, and a dynamically built OpenCV also shows no problems.

I used the following command to configure OpenCV:
`cmake -DBUILD_SHARED_LIBS=OFF -DBUILD_TESTS=OFF -DCMAKE_BUILD_TYPE=Release ..`

If I remove the BUILD_SHARED_LIBS bit from the configuration, the LZW TIFFs are read correctly. Likewise, the homebrew-science version of OpenCV has no problems reading them. Out of curiosity, I attempted to use an external, statically built version of libtiff instead of the OpenCV provided one, but this did not change the behavior.

This problem exists for both OpenCV 2.4.11 and 3.0.0. Am I missing something in my configuration or have I stumbled onto a bug?

**Comment by csparker247** [comment]
Just to update, I've isolated the problem a little more. I have been reading in 16-bit, grayscale TIFFs and using `imread( "image.tiff", CV_LOAD_ANYCOLOR | CV_LOAD_ANYDEPTH)` to load in the unmodified image.

However, still using a static library, if I remove the load flags and just run imread with the image path, the image loads correctly, but downsampled to 8U as expected. If I add back just `CV_LOAD_ANYCOLOR`, the image will also load correctly. Adding back just `CV_LOAD_ANYDEPTH`, however, causes the image not to be read:

```
    cv::Mat test  = cv::imread( imagePath ); // Will load
    cv::Mat test2 = cv::imread( imagePath, CV_LOAD_IMAGE_ANYCOLOR ); // Will load
    cv::Mat test3 = cv::imread( imagePath, CV_LOAD_IMAGE_ANYCOLOR  | CV_LOAD_IMAGE_ANYDEPTH); // Won't load
    cv::Mat test4 = cv::imread( imagePath, -1 ); // Won't load
    cv::Mat test5 = cv::imread( imagePath, CV_LOAD_IMAGE_ANYDEPTH ); // Won't load
```

All of these cases succeed if I'm using the dynamically linked library.

I also tested whether or not the same problem happens if the TIFFs are ZIP compressed, but there were no problems there.

**Comment by alalek** [comment]
Could you please add a minimal sample image (16x16 is fine) to reproduce and confirm this issue?

**Comment by csparker247** [comment]
I'm currently having trouble recreating the issue with the simple imread examples from my last post, however an old build of my full project exhibits the problem with the tif in [this zip](https://github.com/Itseez/opencv/files/71282/00.zip).

Unfortunately, I don't have the time right now to figure out exactly why I'm no longer seeing the behavior I was seeing a month ago. Perhaps you will be able to recreate it with this image. If you cannot, I should be able to look at this again after the New Year and figure out what is required to recreate this.

**Comment by csparker247** [comment]
I no longer believe this to be an OpenCV issue. I've been using CMake on a large project and I had a var `REQ_LIBS` that expanded to all of the lib variables for my dependencies, including OpenCV. If I linked a simple imread test program using `target_link_libraries(cv_test ${REQ_LIBS})` imread would fail on the LZW tif. If, however, I directly linked against OpenCV `target_link_libraries(cv_test ${OpenCV_LIBS})`, the program would work successfully.

I identified that the problem stemmed from including `VTK_LIBRARIES` in my `REQ_LIBS` variable. VTK has a use file for all of its includes and linking, so I'm not sure if what I was doing was just incorrect or if there's a deeper issue with VTK and static OpenCV. Either way, this isn't an OpenCV problem.
